xbmc/addons/AddonDatabase.cpp

```diff
@@ -823,3 +830,48 @@ bool CAddonDatabase::RemovePackage(const std::string& packageFileName)
   return ExecuteQuery(sql);
 }
 
+bool CAddonDatabase::AddSystemAddon(const std::string &addonID)
+{
+  try
+  {
+    if (NULL == m_pDB.get())
+      return false;
+    if (NULL == m_pDS.get())
+      return false;
+
+    if (!IsSystemAddonKnown(addonID)) // Enabled
+    {
+      std::string sql = PrepareSQL("insert into system(id, addonID) values(NULL, '%s')", addonID.c_str());
```

Comment:
@opdenkamp: There is no loop here. But `PrepareSQL()` only makes sense if you also provide the actual value that you want to insert. We do it like this all over the place.